src/lib/Microsoft.Health.Events/EventConsumers/Service/EventBatchingService.cs

```diff
@@ -116,9 +148,16 @@ public async Task ConsumeEvent(IEventMessage eventArg)
 
         private async Task ThresholdCountReached(string partitionId)
         {
-            _logger.LogTrace($"Partition {partitionId} threshold count {_maxEvents} was reached.");
-            var events = await GetPartition(partitionId).Flush(_maxEvents);
+            var partition = GetPartition(partitionId);
+            _logger.LogTrace($"Partition {partitionId} threshold count {partition.GetMaxEventsForPartition()} was reached.");
+            var events = await partition.FlushMaxEvents();
```

Comment:
It appears the partitions only need to know the MaxEvents when you call FlushMaxEvents.  Can we simplify even further over my avoid suggestion by just keep the partition count in the batching service and passing into this method the number of events to flush?  I think that would simplify the code greatly.  This class ends up being the only one that needs to maintain and track the target count.

Reply:
updated to just keep the partition count in the batching service